Update deprecating-or-removing-plugin.adoc (#5786)
jenkins-infra/jenkins.io: content/doc/developer/plugin-governance/deprecating-or-removing-plugin.adoc

Some plugins may be replaced by others or just become irrelevant (e.g. integration with a service which was shut down).
We do not recommend deleting source code outright, even a stale or no-longer-relevant code can still be educational.
However, we do have a mechanism for deprecating or hiding plugins in the Jenkins update centers.

> This page describes the processes for marking a plugin as deprecated, or suspending its distribution entirely.

Reviewer: Minor item, optional.
```suggestion
This page describes the processes for marking a plugin as deprecated or suspending its distribution entirely.
```

== Deprecating a plugin

. Set a `deprecated` label for the plugin that will be visible on plugins.jenkins.io and in the Jenkins plugin manager. This can be done in two ways:
** Put a `deprecated` topic in the plugin's GitHub repository.
   If you have multiple plugins inside a single repository, it will apply to all of them.
   This is the preferred approach.
** Add a `deprecated` label to the plugin entry in the Update Center's link:https://github.com/jenkins-infra/update-center2/blob/master/resources/label-definitions.properties[label-definitions.properties] file.

>    Choose this approach if only some of the plugins in the GitHub repository are to be deprecated.

Author: "delivers" seems odd. Perhaps

```suggestion
   Choose this approach if the GitHub repository contains multiple plugins and only some of the plugins in the repository are to be deprecated.
```

or

```suggestion
   Choose this approach if the GitHub repository hosts multiple plugins and only some of the plugins in the repository are to be deprecated.
```

>    Choose this approach if only some of the plugins in the GitHub repository are to be deprecated.

Reviewer 2: contains multiple sounds better to me

. Update the plugin's documentation to explain the reason of the deprecation.

== Removing a plugin from Update Centers

. Submit a pull request to the Update Center's https://github.com/jenkins-infra/update-center2/blob/master/resources/artifact-ignores.properties[artifact-ignores.properties] file.
  Use the artifact ID as key.
  As value, provide a URL to a web page (usually documentation) that explains to users why distribution is suspended.
  Specifying a URL will also cause a deprecation message to appear.
. Archive the plugin's GitHub repository.
** If you have admin permissions in the repository, it is possible to do it from the GitHub's web interface.

> ** Otherwise, create a link:https://github.com/jenkins-infra/helpdesk/issues/new?labels=github-permissions&template=3-github-permissions.yml[help desk] ticket to archive the plugin's repository.

Reviewer 2: I think it should use the default one is there's no match, the github one asks for different fields and doesn't come close enough to me
```suggestion
** Otherwise, create a link:https://github.com/jenkins-infra/helpdesk/issues/new?labels=triage&template=1-report-issue.yml[help desk] ticket to archive the plugin's repository.
```

> // TODO: Is this the correct link? The template seems inapplicable.

Reviewer: I think that template is sufficient for this rare case.  It doesn't match the task, but the task is so infrequent that creating a template seems more work than is justified.
```suggestion
// TODO: Is this the correct link? The template seems inapplicable.
```

Author: Your suggestion is a no-op I think? Did you mean
```suggestion
```

Reviewer: I did mean that.  Poor use of suggestions on my part.  

I like the suggestion from @timja even better in https://github.com/jenkins-infra/jenkins.io/pull/5786#discussion_r1048178147
